pkg/aspect/clean/clean.go

```diff
@@ -165,3 +189,461 @@ func (c *Clean) Run(_ *cobra.Command, _ []string) error {
+
+	// deleteQueue can add to itself to improve speeds. We need to wait for
+	// deletes to complete before we can close the channel
+	deleteWaitGroup.Wait()
+	close(deleteQueue)
+
+	// sizeQueue will contain all the sizes once the deletes are completed so
+	// we can close the chan before we wait
+	close(sizeQueue)
+	sizeWaitGroup.Wait()
+
+	// all the errors we will receive will be in the errorQueue at this point so we can close the queue before waiting
+	close(errorQueue)
+	errorWaitGroup.Wait()
+
+	if errors.size > 0 {
+		return errors.generateError()
+	}
+
+	return nil
+}
+
+func (c *Clean) confirmationActor(
+	directories <-chan bazelDirInfo,
+	deleteQueue chan<- bazelDirInfo,
+	confirmationWaitGroup *sync.WaitGroup,
+	deleteWaitGroup *sync.WaitGroup,
+) {
+	for bazelDir := range directories {
+		var label string
+		if bazelDir.isCache {
+			label = fmt.Sprintf("Cache: %s, Age: %s, Size: %.2f %s. Would you like to remove?", bazelDir.workspaceName, bazelDir.accessTime, bazelDir.hRSize, bazelDir.unit)
+		} else {
+			label = fmt.Sprintf("Workspace: %s, Age: %s, Size: %.2f %s. Would you like to remove?", bazelDir.workspaceName, bazelDir.accessTime, bazelDir.hRSize, bazelDir.unit)
+		}
+
+		promptRemove := &promptui.Prompt{
+			Label:     label,
+			IsConfirm: true,
+		}
+
+		if _, err := promptRemove.Run(); err == nil {
+			fmt.Printf("%s added to the delete queue\n", bazelDir.workspaceName)
+			deleteWaitGroup.Add(1)
+			deleteQueue <- bazelDir
+		} else {
+			// promptui.ErrInterrupt is the error returned when the user inputs ctrl-c
+			if errors.Is(err, promptui.ErrInterrupt) {
+				// if the user has hit ctrl-c we want to stop prompting them and allow
+				// the program to exit gracefully
+				break
+			}
+		}
+
+		promptContinue := &promptui.Prompt{
+			Label:     "Would you like to continue?",
+			IsConfirm: true,
+		}
+		if _, err := promptContinue.Run(); err != nil {
+			break
+		} else {
+			if errors.Is(err, promptui.ErrInterrupt) {
+				break
+			}
+		}
+	}
+	confirmationWaitGroup.Done()
+}
+
+func (c *Clean) findDiskCaches(
+	sizeCalcQueue chan<- bazelDirInfo,
+	errors chan<- error,
+) {
+	tempDir, err := ioutil.TempDir("", "tmp_bazel_output")
+	if err != nil {
+		errors <- fmt.Errorf("failed to create tmp dir: %w", err)
+		return
+	}
+	defer os.RemoveAll(tempDir)
+
+	bepLocation := filepath.Join(tempDir, "bep.json")
+
+	// Running an invalid query should ensure that repository rules are not executed.
+	// However, bazel will still emit its BEP containing the flag that we are interested in.
+	// This will ensure it returns quickly and allows us to easily access said flag.
+	c.bzl.RunCommand([]string{
+		"query",
+		"//",
+		"--build_event_json_file=" + bepLocation,
+
+		// We dont want bazel to print anything to the command line.
+		// We are only interested in the BEP output
+		"--ui_event_filters=-fatal,-error,-warning,-info,-progress,-debug,-start,-finish,-subcommand,-stdout,-stderr,-pass,-fail,-timeout,-cancelled,-depchecker",
```

Comment:
why bother trying to give a correct and comprehensive set of flags to turn off bazel output, if we throw all the output away anyway? it feels redundant and harder to maintain than just `io.Discard` with no other flags

Reply:
The `io.Discard` does not capture all of the output from the bazel server. If I leave the bazel output on I will still get the following printed to my terminal:
```
ERROR: the empty string is not a valid target
INFO: Build Event Protocol files produced successfully.
```
Even though everything should be discarded.